spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceFeignClient.java

```diff
@@ -36,42 +39,34 @@
  *
  * @since 1.0.0
  */
-final class TraceFeignClient extends FeignEventPublisher implements Client {
+final class TraceFeignClient implements Client {
 
+	private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
 	private final Client delegate;
 	private HttpTraceKeysInjector keysInjector;
+	private final BeanFactory beanFactory;
+	private Tracer tracer;
 
 	TraceFeignClient(BeanFactory beanFactory) {
-		super(beanFactory);
+		this.beanFactory = beanFactory;
 		this.delegate = new Client.Default(null, null);
 	}
 
 	TraceFeignClient(BeanFactory beanFactory, Client delegate) {
-		super(beanFactory);
 		this.delegate = delegate;
+		this.beanFactory = beanFactory;
 	}
 
 	@Override
 	public Response execute(Request request, Request.Options options) throws IOException {
-		Response response;
 		try {
 			addRequestTags(request);
-			response = this.delegate.execute(request, options);
+			Response response = this.delegate.execute(request, options);
+			logCr();
+			return response;
+		} finally {
+			closeSpan();
```

Comment:
If there was an success / exception the span will be closed. Only upon successful response will the span get CR logged